### DISCUSSION TITLE
Video-downloader helper stopped downloading  Fire fox

### DISCUSSION THREAD

**Discussion by JustADarnNiceGuy** [aclap-dev/video-downloadhelper#882] [Help and Q&A]
### Your question

Video downloaded used to work but now will not download. Spins in circles
Firefox v127.0 windows 11 home 64 bit
Video helper 9.0.2.3
Companion app loaded
In settings, if you try to verify companion app it just spins in circles.

 I have uninstalled and reinstalled extension into firefox

For compost and giggles I purchased a license key to see if that would help.
when purchase was complete, it ask to install key.  Key would not take and it gave an error something about not version 7.x.x.x.
I should have written error down. Pretty sure that was the error

If I paste the license key into settings and try to verify, it just spins in circles.

It used us work up on till about a week or 2 ago.

Any suggestions?

### Addon & CoApp version and details

version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: "checking"
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "http_media_download_strategy": "inbrowser",
    "concurrent_downloads_max": 10,
    "show_success_notification_for_icognito": true,
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 2
}

**Comment by JustADarnNiceGuy** [comment]
This is what I mean by "Spinning in circles"
I'm no so sure the companion app actually installed even though it said it did.


![spinning](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/4f671ae0-63ae-4aed-b895-37e6b7752ab0)

**Comment by JustADarnNiceGuy** [comment]
I found a section that ask to check if companion app is installed by checking reg key.
So app should be there.

![companion reg key](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/817aa757-d9ac-4777-8e3b-0eda30af585c)

**Comment by GrampaWildWilly** [comment]
I do not work for VDH.  I'm just another user.

In your problem report, it says this:

> coapp: "checking"
license: {"unset":true}

That says that contrary to appearances, the CoApp did not install correctly.  Your command seems a whole lot more complicated than it really needs to be.  Instead of that, do `dir /s c:\vdhcoapp.exe`, then give us a `dir` of that directory.

On balance, I think it might be wise to uninstall the CoApp.  But don't use Windows tools to do that.  Use Revo Uninstaller.  Do a web search to find that.  Revo does a more thorough job of uninstalling things.  First it executes whatever uninstaller might have been supplied with the product you are uninstalling.  But then it searches for leftovers that the standard uninstaller might have left behind.  Most uninstallers do leave things behind.  So Revo searches the registry for leftovers, then it searches your disk space for leftover files & directories.  You are given the choice at each stage whether to delete the registry keys it finds, & whether to delete the files & directories it finds.  You do have to manually select the items & then tell Revo to actually do the removals.  I rarely second-guess Revo.  But I do always carefully look at what it proposes.  I have been using Revo for years.  It's free for home users.  Truly.  Free.  No strings attached.

So, either close all Firefox windows or disable VDH in the Firefox settings about:addons which you can get from Ctrl+Shift+a.  Then uninstall the CoApp with Revo.

Then try to install the CoApp again.  Maybe it will go better this time.  There are not a whole lot of things you can adjust in the CoApp installer.  I didn't accept the default directory proposed by it but that's purely personal taste.  Just make sure you note what directory the CoApp will be in.  After the installer finishes, go into the CoApp directory & execute this:

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/286dd48c-71ef-4bdb-beab-996eb2149512)

Let's hope you get the same result.  Then relaunch Firefox or re-enable VDH, whichever applies.  Then get into the VDH settings & see what it says.  It should look vaguely like this:

![image](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/0b14fb15-060f-437b-86a6-042b7464ed93)

Also, give us another `dir` of the CoApp directory after what we hope will be a successful install of the CoApp.

If your VDH settings don't look like that, we'll have to investigate some other line of attack.

**Reply by JustADarnNiceGuy** [reply]
I downloaded Revo and gave it a try.
Revo said everything was uninstalled.

Reinstalled co app

cmd prompt give me different reply from coapp than the one you posted but basically said it was good.

Download helper still not working 


![revo uninstall coapp](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/c2052dcb-e46b-4503-a5fe-b1ee358e2fab)
![coapp running](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/4094be84-e4d2-4529-9c0d-eafc465f2031)

**Reply by JustADarnNiceGuy** [reply]
![co app dir](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/f170661f-1af8-41fc-9b38-95acfb783fee)

**Reply by JustADarnNiceGuy** [reply]
But this is interesting

Said could install from command but did not.
Does it thing this is  32BIT windows?

![Uploading cmd install.jpg…]()

**Reply by JustADarnNiceGuy** [reply]
![win64](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/5de8deac-15f7-45ac-b952-42cb990fa70b)
![cmd install](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/4dd0174d-6eb1-4c72-93f4-fc7d7d0bf672)

**Reply by GrampaWildWilly** [reply]
Well, first of all, the command I issued had a parameter: `--version`.  You need to execute `vdhcoapp` with that parameter.  Look again closely at the command window image I posted.

Win32?  But your system info says you are running a 64-bit Windows.  That might mean you just picked the wrong CoApp from the Wiki page.  Make sure you are downloading the right CoApp.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/ed48f246-0279-4424-8e7c-96562e2fa942)

I think you should start over.  Download the right CoApp.  Close all your browser windows.  Revo Uninstall the CoApp.  Install the fresh copy of the CoApp you just downloaded.  Read the next paragraph before proceeding.  Run the command I showed earlier at the command line with the `--version` parameter.  There's 2 dashes there.   Make sure that's what you are executing.  If it does not give you the result of 2.0.19, something is still wrong & there's no point in even going any further.  Assuming you now have 2.0.19 installed, then reopen the browser & check the VDH settings.

Do not attempt to install the CoApp from the command line.  Go into File Explorer.  Navigate to the directory into which you downloaded the CoApp installer executable.  Use whatever combination of PageUp, PageDown, UpArrow, DownArrow it takes to get that .exe file highlighted.  Then hit Enter.

That's the installer.  The installer is not the CoApp itself.  The CoApp is `vdhcoapp.exe`.  The only way to execute `vdhcoapp.exe` is from the command line, & you should only ever supply the parameter `--version`.  The CoApp is not a user application.  You should never execute it yourself, neither from the command line nor from File Explorer, except in this one unusual case with the `--version` parameter.  After you do the `--version` thing, you should NEVER need to execute the CoApp from the command line again.  This `--version` thing is a one-time verification that you installed the CoApp properly.  After you do that just this once, you don't need to ever again run it.  You've verified it's installed & you can rely on that from now on.  And executing it from File Explorer is something you should NEVER do, period.

**Reply by GrampaWildWilly** [reply]
I want to make sure I am very clear about this.  The thing you download from the Wiki is the CoApp installer.  You execute that from the File Explorer.  It unpacks some files & places them where you say.  Among those files it unpacks is `vdhcoapp.exe`.  That is the CoApp.  Note the terminology.  CoApp installer.  CoApp.  Two different things.  You execute the installer from the File Explorer.  That should be necessary only once.  You will execute the CoApp itself only ever once.  That will be from the command line, not File Explorer.  That one execution will supply the `--version` parameter.  This is to verify that it's properly installed.  You will never need to do this but just this once.  Forever after, the only way the CoApp gets executed is from VDH while you are doing a download.  You will NEVER execute the CoApp yourself except for just this once.

**Reply by JustADarnNiceGuy** [reply]
Download helper used to work, stopped maybe 2 weeks ago.  Do not use it very often.

Firefox added the extension, Downloadhelper.
When you try to download a video, downloadhelper says it needs Co app to run.
Downloader provides a link to co app.
Click the link and download.
Click the down load and it installs.

I do have Comodo antivirus installed but i tell it to let co app run unlimited.
after it failed, I tried Revo.
ran the uninstall inside Co app and Revo  said it was all gone.

I turned off comodo antivirus, ( i hate running with out antivirus but need to figure out what's going on)
open firefox, added downloadhelper extension.
Tried to download, download helper said it needed co app.
Clicked the link downloadehelper provided and downloaded co app.
Double clicked the downloaded co app installer
Co app installed and seemed happy
Tried to download and no go.

I see I missed to add the -- before version. that is why it gave me the different reply about running successfully.

vdhcoapp --version will return 2.0.19


I'm going to be out of cell service range for the next day.
Why I like downloader can download stuff to watch while in no cell land.

**Reply by GrampaWildWilly** [reply]
I'm not sure this is significant but personally, I wouldn't do things in the order you've done them.  I would do this:

1. Close all browser windows.
2. Revo uninstall the CoApp . . . yeah I know, yet again.  Make sure all browser windows are closed when you do this.
3. Open your browser & navigate to the Wiki.  In the Wiki, find the CoApp installer.  I have an image above showing what the relevant bit of that page looks like.  Download the CoApp installer from the Wiki.  Do not do the download from any link provided during an attempted VDH download.  Download it from the Wiki, without any involvement of the VDH add-on.
4. Close all browser windows.  Again.
5. Still with all browser windows closed, execute the CoApp installer you downloaded from the Wiki.
6. Just to reassure yourself (and me), run `vdhcoapp --version` & prove you got CoApp 2.0.19 . . . yet again.
7. Now, finally, reopen the browser & go into the VDH settings.  Compare what you have in the `More settings` section with what I posted upthread here.  In fact, post a screenshot of that page of VDH settings.  You should not have any license information there.  If you do, click the little white x in the gray spot beside your license number so your screenshot does NOT display your license number.

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/dc862f11-7d75-43e1-91dd-0eff2dc461db)

This is making me crazy.  I'm sure it's making you crazier.

**Reply by JustADarnNiceGuy** [reply]
Got back on this side of mountain sooner than I thought.

I greatly appreciate your help.

Yea, I understand about going crazy. Many Moons ago when I was a much younger man I used to be a desktop tech.
Now I drive myself crazy in the autonomous vehicle field. 

I'll give it another try. follow your install and I will document what happens on this end
Kinda wondering if firefox took a dump. I think firefox ver was 126.0     I hate the x.0 version of anything

Spin the wheel, no whammies......

**Reply by JustADarnNiceGuy** [reply]
good think I don't have any hair left.

1) made sure all browsers were closed.
2) Open Revo Pro, selected Coapp uninstall, ran the uninstall inside Revo
ran the scan, said one key not deleted ( uninstall key)  deleted key and folder
Said would delete on next restart. shut down and restarted computer.
3) open browser - found Wiki - downloaded the 64 bit
4) Closed browser
5) ran the co app installer (double clicked)
6) Said it installed and life was good
7)  Command prompt vdhcoapp --version  returned  2.0.1.9
8) open Browser go to downloadhelp - options - more    does not verify co app installed - spins in circles
9) closed browser
For compost and Giggles
Disabled Anti virus
Disabled fire wall
Disabled web filtering
10) open browser - download helper - option - more.   Will not verify co app

I attached screen grabs as I went.  Maybe I did something really stupid
Not sure the order forum will post them in.

I have not tried to enter my license.

I do greatly appreciate your help because this should be a simple fix but kicking my ass.



Copied setting:
version: 9.0.2.3
target: mozilla
channel: stable
lang: en-US
coapp: "checking"
license: {"unset":true}
platform: x86-64 win
UA: Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:127.0) Gecko/20100101 Firefox/127.0
{
    "view_options": {
        "all_tabs": false,
        "low_quality": false,
        "sort_by_status": true,
        "sort_reverse": false,
        "show_button_clean": true,
        "show_button_clean_all": false,
        "show_button_convert_local": false,
        "hide_downloaded": false
    },
    "open_count_store": 2
}



![revo uninstall coapp again 7](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/2a4c270f-e081-4062-9bdc-93d2e1ba0079)
![revo uninstall coapp again 6](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/d3cc6ef0-dc0b-4c78-9b66-25a658d52928)
![revo uninstall coapp again 5](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/57964100-2b64-4aa2-bec3-3c0432f6ee87)
![revo uninstall coapp again 4](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/a1948c56-dde9-4217-bc8a-4cd7c4951c50)
![revo uninstall coapp again 3](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/1b860e3f-12dc-439f-bd99-7e935904bc4b)
![revo uninstall coapp again 2](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/9a153568-c73d-4a84-aa07-f39828c969e3)
![revo uninstall coapp again 1](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/f154cf4a-af56-4db0-b496-050b65b995f7)

**Reply by JustADarnNiceGuy** [reply]
posted pics in reverse order

**Reply by GrampaWildWilly** [reply]
I agree.  This should be simple.  I can't tell from what you've posted that anything you've done is wrong.  When I did it, it just worked.  A few months ago I built a new computer.  Virgin hardware.  No software.  Installed W11 Pro.  Installed Firefox.  Installed VDH.  Installed the CoApp.  Everything just worked.  I've never encountered a problem like what you're having.  This is now officially beyond anything that I, as just another user, can diagnose.  I think we're going to need @paulrouget Paul's help here.  There is a chapter in the Wiki about turning on logs.  But I'm not sure exactly at what point you're supposed to do that.  And I sure won't claim to be able to interpret the logs, although there's always the chance something obvious will be in there.  But I think we need Paul's guidance now.

**Reply by GrampaWildWilly** [reply]
Failing Paul's help in a timely manner, I suppose the next step is to uninstall Firefox itself.  This would be perhaps somewhat painful.  127.0.1 just came out today so I think I would do things in this order.

Before we start, I think the Firefox Sync feature is very useful.  When I did that W11 install, followed by the Firefox install, all I had to do was log into my Firefox account & tell it to sync.  Bam, I had all my bookmarks.  I think it also installed all my add-ons.  My memory is already getting fuzzy on this & it was barely 6 months ago.  But I think the sync operation gave me VDH back again without my even having to install it myself.  In light of that, I would uninstall VDH & then do a sync so your synced data would not include VDH.  Sync does require some configuring so do spend some time looking at the Sync configuration & thinking about what you want to Sync.  There is a Sync section in the Firefox settings where you configure what to sync.  In my case, I am syncing bookmarks, history, & settings.  But you need to make your own choices.  This also, by the way, syncs with Firefox on my cell phone, but that's irrelevant here.

I would also get on the Firefox web site & download the 127.0.1 installer.  That's not a 100% obvious process so . . .

https://www.mozilla.org

![#01](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/897aa452-844e-49be-aaca-a5c066ca6425)
![#02](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f02a3641-bbde-4a83-8e6c-234a87bebbda)
![#03](https://github.com/aclap-dev/video-downloadhelper/assets/75588964/f81a80f7-fff1-4f79-bb35-5076a8c00867)

That will download the right installer executable, which will make things a lot simpler later on.  I do this exact process every time there's a Firefox update.  I keep the current installer handy just in case.  You never know when it will be useful.

So here's my recipe.  Analyze it yourself carefully.  I have not personally done this but this is how I would do this.

1. Uninstall VDH from the about:addons page.  Disable just for good measure, then Remove.
2. Sync Firefox.
3. Close all Firefox windows.
4. Revo uninstall Firefox.  By the way, I'm impressed you bought Revo Pro.  I've only ever used the free one all these years.
5. I don't know if Revo will tell you that a reboot is required.  Most of the time when I've Revoed anything, it doesn't tell me to reboot.  Once in a while, but not usually.  However, if that turns out to be the case, well, so be it.
6. Execute the Firefox 127.0.1 installer you downloaded earlier.
7. Sync Firefox to get back your old configuration.
8. Come into the Wiki here & install VDH.  I would recommend the beta.  The beta includes bug fixes.  There's no sense in opening yourself up to hitting bugs that Paul has already corrected.  It's your choice, but since you would be starting fresh here, may as well get the beta.
9. Run your CoApp installer, which you already have from all this previous folderol.
10. Pray to the Cybergods this works.

**Reply by JustADarnNiceGuy** [reply]
People put a lot of work and effort into writing these programs.
I figure if I use it and it is useful, the few bucks it cost will not kill me and it is incentive for them to keep improving the programs.
I know a lot of the guys who write free ware do it for the love and challenge of it. 

I'm up in sticks of Alaska right now, Fly back to lower 48 end of month.
Dont want to try to undo firefox up here. Internet bites at remote sites.

For compost and giggles, I added Download helper extension in chrome. 
I assumed being Co App was already installed Chrome downloader helper would find Co App.
Co App says it is the same for all browser but downloaderhelp is browser specif.
Chrome downloadhelper is not finding Co App
recheck just search and returns.  Wants you to download and install co app

I think my comodo decided to do something dumb. like block it somewhere.  Even though I told it to give total access.
Thinking Comodo black listed it.
Will have to dig thru comodo logs.

**Reply by GrampaWildWilly** [reply]
I used Comodo for several years on W7.  I was fully prepared to install it on W11 when I built my new system.  But when I discovered there was already an antivirus built into W11, I decided to hold off & see how it worked.  It worked OK as far as I could see.  Virus signatures are sent out several times a day.  It's never squawked about anything to do with VDH/CoApp.  And then Paul made his comment in the Wiki about antivirus.  It might give you pause.

I found that Comodo was usually annoying but it didn't actually interfere with anything.  The annoying bit went like this.  I have certain .bat scripts I execute all the time.  One such is the one I run ffprobe from, usually in conjunction with trying to help somebody here.  I replace the URL of the manifest in question inside the script, save it, then run it.  Every single stinking lousy time, Comodo would log a warning about how it didn't recognize the file, but it let it run anyway.  Same file, over & over & over.  I can't possibly count how many times I told it my ffprobe script was a known safe object.  Didn't matter.  It flagged it every single time.  I tried marking the directory containing the script as safe.  Didn't matter.  I never did find a way to make Comodo ignore that script.  And there were others.  At least it let the script run.  The first time I saw the warning, I must have had 300 entries in the log complaining about that file.  That taught me I needed to clear the log more often.  I ended up running the script & immediately going into Comodo to clear the inevitable message that would appear.  I kept the Comodo window open permanently just to deal with this.  So if Comodo really is interfering with your operations, I would advise you to simply uninstall it.  That is also Paul's advice in the Wiki.  Just let Windows Defender carry the load.

But like I say, Comodo never actually interfered with anything.  The closest it ever came was on occasion when I was installing something new, it would pop up a nag screen & I could tell it no, this thing is OK, let it run.  And it would run & I wouldn't have any static from Comodo after that.  I don't think I ever spent a whole lot of time configuring it so its default behavior was pretty close to acceptable.  The one thing I liked about it was that it has this really nice performance monitor buried way inside it.  I don't even remember any more how to get into it.  I just stumbled on it one day & after that, I would open it every so often when I wanted to monitor something even more closely than the Resource Monitor did.  I can't say I've missed it since I got on W11.  The Resource Monitor is still here & it's quite good enough.

So I ran VDH/CoApp on W7 with Comodo for several years & never had any kind of problem even close to what you're seeing.  Mind you, that was VDH 7 & 8 plus early incarnations of CoApp 2.  Still, I'd be surprised if it would do anything to the modern versions.

I suppose it's circumstantial evidence that Comodo might be interfering if Chrome is giving you the same problem now.  This all certainly has me stumped.

Don't get frostbite or, more likely at this time of year, black fly bitten to death.  When you return to civilization, we can pursue this.  Or maybe Paul will have some brilliant idea & help us out here before then.

**Reply by JustADarnNiceGuy** [reply]
I have a contractor showing up tomorrow so not going to have time to play with this till get back into the states.

Been running Comodo for a long time, it has it issues but seems like the best out there.

I did run a quick test because Comodo did bite me in the ass years ago.  Comodo has a Sandbox and now they have a "Auto-Containment"

I do not remember when the "Auto-Containment" was added but I do know VDH stopped working with the new upgrade.
only reason I noticed VHD upgrade was the interface.  I wonder if something changed in the core program of VDH or just the interface.

VDH would fail in Chrome also.  If I tun off "anti-containment"  VDH will work in Chrome but not Firefox.  Which really points me to something in Comodo.
Auto-contain enable- No Coapp found
Auto-contain dis-abled - finds Coapp in Chrome but not Firefox

When running VDH in Chrome, it will start a download than pops a box that says " Download failed, sadly we failed downloading that specific media"  Does that on every site that used to work.  Granted I have never tried VHD in Chrome so hard to say if it worked before .

I tried to run Chrome and Firefox as administrator (even though my account has full admin rights, there is a difference) same things.

Years ago, comodo bit me in the ass with a program called HMI Droid.  A Human Machine Interface program.
I wrote a program in "Click" to control a 3 wheel, 3 axis pitching machine, and 2 stepper motors with feedback for ball feed. I prefer "Allen Bradly" but they are so dam expensive.   I used HMI Droid as the interface to an Android Tablet so the baseball coach could control the pitching machine from anywhere.  Coach can stand by the player and coach while also controlling the pitch, speed and placement of ball.

Long story short, I did all the HMI coding was was going to export to Android tablet.  NOTHING was there... instant panic because the program, all archives and all backup were only on the laptop.  Holy Crap.....
Reboots cures a lot but all files were missing.
Comodo did something and hid all the files.  I had to run HMI Droid as admin and files were there.....
Why I tried Chrome and Firefox run as Admin.

I attached screens of the Bored Grandpa in the Garage pitching machine screens.
Allows coach to control 3 wheel pitching machine remotely from Android tablet.
Can control type of pitch
control X,Y,Z axis of pitch placement.
Steppers control ball feed into machine. Holds 25 balls
Adding a fake arm to simulate the pitchers wind up.
![HMI Droid 1](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/55596cda-7aae-47bf-8996-02410a7a54de)
![HMI Droid 2](https://github.com/aclap-dev/video-downloadhelper/assets/172988376/46f0c747-5b73-4393-97ee-82fab1b9ebf7)

**Comment by paulrouget** [comment]
I didn't read the whole conversation, but do you mind trying this: https://github.com/aclap-dev/video-downloadhelper/wiki/How-to-enable-logs

**Reply by GrampaWildWilly** [reply]
For comparison, here's what I did.

1. Followed instructions for enabling logs.
2. Opened the `More settings` page of VDH settings.
3. Stopped the log & saved it.

And here's the log I got.

[vdh-logs.txt](https://github.com/user-attachments/files/15908143/vdh-logs.txt)

This is what it looks like when everything works right & VDH recognizes the CoApp.

**Comment by JustADarnNiceGuy** [comment]
Hi,
First off, I'd like to thank you for all the help.
I'm back in the states and took some time to try again.

Just a quick recap if someone has the same issues.
Download helper was working ( for months) and suddenly stopped working.

I ran through the uninstall with Revo again.
I noticed there was a McAfee app in the list. ( this computer came with McAfee when purchased about a year ago)
I did an uninstall of McAfee when I purchased the computer because I like Comodo and have it licensed.

For some reason, McAfee must have reactivated itself and was blocking Video download helper.
It was not the full blown version of McAfee but a Browser app. ( I should have done a screen grab)

Used Revo to uninstall McAfee.
Removed DVH extension from FireFox
Revo to uninstall coapp
installed DVH
installed Coapp
VHD is working again in fireFox.

Wish I had notice McAfee in the list when this problem first started to happen.

**Reply by GrampaWildWilly** [reply]
Son of a . . .

When CCleaner updates itself, it always prompts me to install . . . I forget now what it is.  But it's something I don't want.  I have to be careful to not accidentally install it.  Maybe some entirely unrelated installation you did had a hidden oh by the way we will also be oh so nice to you & install this thing you have already expressed a strong desire to avoid.  You're welcome.  If this is what AI gives us, the intelligence is indeed artificial.  This sort of stealth-install is a very common blight on our computer experience, as I'm sure you well knew before this came up.  You can just never let down your guard.  It's like the multiple times a day Microsoft tries to convince me there's a problem with my account.  There's no problem.  The problem is them telling me there's a problem when there isn't one.

Come to think of it, Comodo used to nag me about switching to their browser.  (Is it called Dragon?  It's been long enough I've forgotten.)  They also kept trying to change my default search engine.  I do so much not miss all that.

It's typical that you accidentally noticed it.  I can't tell you how many times I've accidentally noticed something that led to great advances in my knowledge & understanding of things.  I'm just glad you did finally get to the bottom of this.  Thank you for keeping the forum posted on your escapades.
